### PR TITLE
Update PeerManager to new storage system

### PR DESCRIPTION
**Pull request by neonknight64**
## Description
- Update the PeerManager and PeerStorage to make use of the new storage system
- Had to change each test that makes use of the PeerManager to allow them to create and clean the peer database
- The PeerManagerError has been moved to error.rs

## Motivation and Context
The PeerManager is more complex than needed, this complexity was introduced to circumvent the limitations of the old storage system.

## How Has This Been Tested?
New tests have been added and existing tests have been modified.

## Types of changes

* [x] Bug fix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [x] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:

* [x] I'm merging against the `development` branch
* [x] I ran `cargo-fmt --all` before pushing
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

File: base_layer/p2p/tests/data/README.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Temp folder for LMDB database files
```
Comment (sdbondi) [inline base_layer/p2p/tests/data/README.md]: Perhaps the database files should be gitignored incase they leak - I'd rename the readme to .gitkeep which is the idiomatic way to keep an 'empty' directory in the repo in git

File: comms/src/inbound_message_service/inbound_message_worker.rs [hunk 2]
```diff
@@ -255,6 +256,22 @@ mod test {
     #[test]
     fn test_dispatch_to_multiple_service_handlers() {
         init();
+
+        // Clear and setup DB folders
+        let test_dir = "./tests/test_peer_storage";
+        let database_name = "peer_manager";
+        if std::fs::metadata(test_dir).is_ok() {
```
Comment (sdbondi) [inline comms/src/inbound_message_service/inbound_message_worker.rs]:
```suggestion
        if fs::metadata(test_dir).is_ok() {
```

File: comms/src/inbound_message_service/inbound_message_worker.rs [hunk 3]
```diff
@@ -389,5 +406,12 @@ mod test {
         std::thread::sleep(time::Duration::from_millis(200));
         thread_handle.join().unwrap();
         assert!(client_connection.send(message1_frame_set).is_err());
+
+        // Clear up DB folders
+        let _no_val = fs::remove_dir_all(test_dir);
+        if std::fs::metadata(test_dir).is_ok() {
```
Comment (sdbondi) [inline comms/src/inbound_message_service/inbound_message_worker.rs]:
```suggestion
        if fs::metadata(test_dir).is_ok() {
```

File: base_layer/p2p/tests/data/README.md (hunk 1 above)
Comment (neonknight64) [reply base_layer/p2p/tests/data/README.md]: That is cleaner, I applied the change.

File: comms/src/inbound_message_service/inbound_message_worker.rs (hunk 2 above)
Comment (neonknight64) [reply comms/src/inbound_message_service/inbound_message_worker.rs]: Change applied

File: comms/src/inbound_message_service/inbound_message_worker.rs (hunk 3 above)
Comment (neonknight64) [reply comms/src/inbound_message_service/inbound_message_worker.rs]: Change applied

**Pull request merged**